lib/dentaku/ast/comparators.rb

```diff
@@ -28,7 +28,7 @@ def value(context = {})
 
       def cast(val)
```

Comment:
I've noticed very similar logic within [`Dentaku::AST::Arithmetic#numeric`](https://github.com/rubysolo/dentaku/blob/0b841e962bd141afe3d3023c630c1571b08ebaff/lib/dentaku/ast/arithmetic.rb#L48). Should this logic be encapsulated somewhere and reused here instead of fixing corner cases? :thinking:

Reply:
how could we trigger that exception in a spec for Arithmetic? do you have a spec example?

Reply:
I'm not sure it could be triggered, not that familiar with the Dentaku, sorry ":D I just would like to suggest tweaking `cast` regular expression instead of adding each corner case. Maybe just copy-pasting part of Arithmetic logic :thinking: 
```
      def cast(val)
        return val unless val.is_a?(::String)
        # NOTE: we also get rid of val.empty?, since this corner case doesn't match numerical regular expresions

        case val.to_s
        when /\A-?\d*\.\d+\z/ then BigDecimal(val, Float::DIG + 1) # NOTE: added `-?` here. Arithmetic misses it for whatever reason
        when /\A-?\d+\z/ then val.to_i
        else val
        end
      end
```